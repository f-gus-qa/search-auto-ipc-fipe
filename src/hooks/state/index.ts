export { default as useAutoState } from './useAutoState';
export { default as useLoadingState } from './useLoadingState';
export { default as useSelectState } from './useSelectState';
export { default as useAlertState } from './useAlertState';
