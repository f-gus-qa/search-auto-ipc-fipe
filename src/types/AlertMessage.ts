import type { VariantType } from 'notistack';

interface AlertMessage {
  message: string,
  type: VariantType,
}

export default AlertMessage;
