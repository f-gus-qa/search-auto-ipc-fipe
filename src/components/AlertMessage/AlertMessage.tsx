import { useEffect } from 'react';
import { Button } from '@material-ui/core';
import { useSnackbar } from 'notistack';
import { useAlertState } from '@/hooks/state';

function AlertMessage() {
  const alert = useAlertState();
  const { enqueueSnackbar, closeSnackbar } = useSnackbar();

  useEffect(() => {
    if (alert) {
      enqueueSnackbar(alert.message, {
        variant: alert.type,
        anchorOrigin: {
          horizontal: 'center',
          vertical: 'top',
        },
        action: (key) => (
          <Button
            variant="text"
            size="small"
            onClick={() => closeSnackbar(key)}
          >
            fechar
          </Button>
        ),
      });
    }
  // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [alert]);

  return <></>;
}

export default AlertMessage;
