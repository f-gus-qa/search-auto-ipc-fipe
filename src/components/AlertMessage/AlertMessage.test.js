import React, { useEffect } from 'react';
import { render } from '@testing-library/react';
import { create } from 'react-test-renderer';
import { omit } from 'lodash';

import * as notistack from 'notistack';
import { useAlertState } from '@/hooks/state';
import AlertMessage from './AlertMessage';

const enqueueSnackbar = jest.fn();

const alert = {
  type: 'success',
  message: 'some-message',
};

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useEffect: jest.fn().mockImplementation((f) => f()),
}));

jest.mock('@/hooks/state', () => ({
  useAlertState: jest.fn(),
}));

useAlertState.mockImplementation(() => alert);

describe('<AlertMessage>', () => {
  jest.spyOn(notistack, 'useSnackbar').mockImplementation(() => ({
    enqueueSnackbar,
    closeSnackbar: jest.fn(),
  }));

  it('checks tree', () => {
    const { toJSON } = create(<AlertMessage />);
    expect(toJSON()).toMatchSnapshot();
  });

  describe('when renders', () => {
    beforeEach(() => {
      render(<AlertMessage />);
    });

    it('checks effect when rerender', async () => {
      expect(useEffect).toHaveBeenCalledTimes(1);
    });

    it('checks dispatch alert', () => {
      const [arg0, arg1] = enqueueSnackbar.mock.calls[0];

      expect(arg0).toEqual(alert.message);
      expect(typeof arg1.action).toBe('function');
      expect(omit(arg1, ['action'])).toEqual({
        variant: alert.type,
        anchorOrigin: {
          horizontal: 'center',
          vertical: 'top',
        },
      });
    });
  });
});
