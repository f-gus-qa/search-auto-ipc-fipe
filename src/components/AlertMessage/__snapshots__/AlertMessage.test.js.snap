// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertMessage> checks tree 1`] = `null`;
